src/fcntl.rs

```diff
@@ -255,182 +255,53 @@ mod consts {
-            const O_TRUNC     = libc::O_TRUNC;
-            const O_APPEND    = libc::O_APPEND;
-            const O_NONBLOCK  = libc::O_NONBLOCK;
-            const O_DIRECTORY = 0x0020000;
-            const O_NOFOLLOW  = libc::O_NOFOLLOW;
-            const O_CLOEXEC   = libc::O_CLOEXEC;
-            const O_SYNC      = libc::O_SYNC;
-            const O_NDELAY    = libc::O_NDELAY;
-            const O_FSYNC     = libc::O_FSYNC;
-            const O_SHLOCK    = 0x0000080;
-            const O_EXLOCK    = 0x0000020;
-            const O_DIRECT    = 0x0010000;
-            const O_EXEC      = 0x0040000;
-            const O_TTY_INIT  = 0x0080000;
-        }
-    );
-
-    bitflags!(
-        pub struct FdFlag: c_int {
-            const FD_CLOEXEC = 1;
-        }
-    );
-}
-
-#[cfg(target_os = "openbsd")]
-mod consts {
-    use libc::{self, c_int};
-
-    bitflags!(
+    libc_bitflags!(
         pub flags OFlag: c_int {
-            const O_ACCMODE   = libc::O_ACCMODE,
-            const O_RDONLY    = libc::O_RDONLY,
-            const O_WRONLY    = libc::O_WRONLY,
-            const O_RDWR      = libc::O_RDWR,
-            const O_CREAT     = libc::O_CREAT,
-            const O_EXCL      = libc::O_EXCL,
-            const O_NOCTTY    = libc::O_NOCTTY,
-            const O_TRUNC     = libc::O_TRUNC,
-            const O_APPEND    = libc::O_APPEND,
-            const O_NONBLOCK  = libc::O_NONBLOCK,
-            const O_DIRECTORY = 0x0020000,
-            const O_NOFOLLOW  = libc::O_NOFOLLOW,
-            const O_CLOEXEC   = libc::O_CLOEXEC,
-            const O_SYNC      = libc::O_SYNC,
-            const O_NDELAY    = libc::O_NDELAY,
-            const O_FSYNC     = libc::O_FSYNC,
-            const O_SHLOCK    = 0x0000080,
-            const O_EXLOCK    = 0x0000020,
+            O_ACCMODE,
+            O_RDONLY,
+            O_WRONLY,
+            O_RDWR,
+            O_NONBLOCK,
+            O_APPEND,
+            O_SHLOCK,
+            O_EXLOCK,
+            O_ASYNC,
+            O_SYNC,
+            O_NOFOLLOW,
+            O_CREAT,
+            O_TRUNC,
+            O_EXCL,
+            O_NOCTTY,
+            O_DIRECTORY,
+            O_CLOEXEC,
+            O_FSYNC,
+            O_NDELAY,
+            #[cfg(any(target_os = "netbsd", target_os = "openbsd", target_os = "macos", target_os = "ios"))]
+            O_DSYNC,
+            #[cfg(any(target_os = "netbsd", target_os = "dragonfly", target_os = "freebsd"))]
+            O_DIRECT,
+            #[cfg(any(target_os = "netbsd", target_os = "openbsd"))]
+            O_RSYNC,
+            #[cfg(target_os = "freebsd")]
+            O_EXEC,
+            #[cfg(target_os = "freebsd")]
+            O_TTY_INIT,
+            #[cfg(target_os = "netbsd")]
+            O_ALT_IO,
+            #[cfg(target_os = "netbsd")]
+            O_NOSIGPIPE,
+            #[cfg(target_os = "netbsd")]
+            O_SEARCH,
         }
     );
 
-    bitflags!(
+    libc_bitflags!(
```

Comment:
Some of linux flags are not in `libc` yet, so I can't use `libc_bitflags!` for them now.
This makes it impossible to merge all `consts` into one.